KAFKA-15949: Unify metadata.version format in log and error message

FrankYang0529 opened the pull request:
There were different words for `metadata.version` like `metadata version` or `metadataVersion`.
Unify format as `metadata.version`.

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

showuon commented on core/src/main/scala/kafka/tools/StorageTool.scala:
```diff
@@ -139,7 +139,7 @@ object StorageTool extends Logging {
     formatParser.addArgument("--release-version", "-r").
       action(store()).
-      help(s"A KRaft release version to use for the initial metadata version. The minimum is 3.0, the default is ${MetadataVersion.LATEST_PRODUCTION.version()}")
+      help(s"A KRaft release version to use for the initial metadata.version. The minimum is ${MetadataVersion.IBP_3_0_IV0}, the default is ${MetadataVersion.LATEST_PRODUCTION.version()}")
```
I saw 2 formats: `MetadataVersion.IBP_3_0_IV0` and `MetadataVersion.IBP_3_0_IV0.version()` here. Could you help me understand what output it will be for each of them? And I think we should make them consistent. Same as other places.

FrankYang0529 replied:
Hi @showuon, thanks for the review. Using `MetadataVersion.IBP_3_0_IV0` or `MetadataVersion.IBP_3_0_IV0.version()` has same result. Both `toString` and `version` functions return `ibpVersion`:

https://github.com/apache/kafka/blob/58ddd693e69599b177d09c2e384f31e7f5e11171/server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java#L523-L525
https://github.com/apache/kafka/blob/58ddd693e69599b177d09c2e384f31e7f5e11171/server-common/src/main/java/org/apache/kafka/server/common/MetadataVersion.java#L632-L634

showuon replied:
OK, thanks for the explanation. Could we at least make them consistency (all with `.version()` or all without it)?

FrankYang0529 replied:
Yes, it looks like we only use `.version()` in few messages, so I remove it. Thank you.

showuon reviewed (commented):
Thanks for the PR. Left some comments.

showuon reviewed (approved):
LGTM!

showuon commented:
Retriggering the CI  build: https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-15505/4/

showuon commented:
@FrankYang0529 , there are some tests failing because of your change. Please take a look. Thanks.

https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-15505/4

FrankYang0529 committed:
KAFKA-15949: Unify metadata.version format in log and error message

There were different words for metadata.version like metadata version
or metadataVersion. Unify format as metadata.version.

Signed-off-by: PoAn Yang <payang@apache.org>

FrankYang0529 commented:
> @FrankYang0529 , there are some tests failing because of your change. Please take a look. Thanks.
> 
> https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-15505/4

Sorry for that. I will double check the test result next time. I have fixed error in `testScramWithBadMetadataVersion`, `testMetadataVersionChangeExceptionToString`, and `testCreateClusterInvalidMetadataVersion`. Thank you.

showuon commented:
Re-running CI build: https://ci-builds.apache.org/job/Kafka/job/kafka-pr/job/PR-15505/6/

showuon commented:
Failed tests are unrelated.
